roles/cifmw_cephadm/tasks/configure_object.yml

```diff
@@ -38,21 +38,97 @@
+    - cifmw_ceph_daemons_layout.rgw_enabled | default(true) | bool
+    - not swift_in_ctlplane.stdout | bool
+    - swift_endpoints_count.stdout == "0"
+
+- name: Capture required UUIDs of swift user/service
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+  delegate_to: localhost
+  when: cifmw_openshift_kubeconfig is defined
+  block:
+    - name: Get uuid for user 'swift'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show swift -c id -f value"
+      register: user_swift_uuid
+
+    - name: Get uuid for service 'swift'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack service show swift -c id -f value"
+      register: service_swift_uuid
+
+    - name: Get uuid for role 'member'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show member -c id -f value"
+      register: role_member_uuid
+
+    - name: Get uuid for role 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show admin -c id -f value"
+      register: role_admin_uuid
+
+    - name: Get uuid for project 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack project show admin -c id -f value"
+      register: project_admin_uuid
+
+    - name: Get uuid for user 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show admin -c id -f value"
+      register: user_admin_uuid
+
+    - name: Get uuid for role 'swiftoperator'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show swiftoperator -c id -f value"
+      register: role_swiftoperator_uuid
+
+- name: Check if cifmw_cephadm_certificate path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_certificate }}"
+  register: cert_stat_result
+  when: cifmw_cephadm_certificate | length > 0
+
+- name: Check if cifmw_cephadm_key path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_key }}"
+  register: key_stat_result
+  when: cifmw_cephadm_key | length > 0
+
+- name: Update urischeme based on cert/key
+  ansible.builtin.set_fact:
+    cifmw_cephadm_urischeme: "https"
+  when:
+    - cifmw_cephadm_certificate | length > 0
+    - cifmw_cephadm_key | length > 0
+    - cert_stat_result.stat.exists | bool
+    - key_stat_result.stat.exists | bool
+
+- name: Configure object store to use rgw
+  cifmw.general.ci_script:
+    extra_args:
+      KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    output_dir: "/home/zuul/ci-framework-data/artifacts"
+    script: |-
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user {{ user_swift_uuid.stdout }} --project {{ project_service_uuid.stdout }} {{ role_member_uuid.stdout }}
```

Comment:
I'm wondering if `role add --user` can be separated from `endpoints` commands. I'm ok to do that in a follow up.
The reason is that we might have a VIP for public, and a VIP for internal endpoint.

Reply:
We can address this by having two variables: public VIP and private VIP. on lines 129 and 130 I'll add  that later.